vault/cluster/cluster.go

```diff
@@ -438,7 +440,23 @@ func (cl *Listener) GetDialerFunc(ctx context.Context, alpn string) func(string,
 		tlsConfig.NextProtos = []string{alpn}
 		cl.logger.Debug("creating rpc dialer", "address", addr, "alpn", alpn, "host", tlsConfig.ServerName)
 
-		return cl.networkLayer.Dial(addr, timeout, tlsConfig)
+		conn, err := cl.networkLayer.Dial(addr, timeout, tlsConfig)
+		if err != nil {
+			return nil, err
+		}
+		logTLSConnection(cl.logger, conn.ConnectionState())
```

Comment:
Can we wrap this in a `if cl.logger.IsTrace() {}` branch to avoid doing this extra work all the time?

Reply:
Good idea